File: LeetCode/Simulation/Golang/Leetcode2367-number-of-arithmetic-triplets.go
package sbgo

func arithmeticTriplets(nums []int, diff int) int {
	mp := map[int]bool{}
	for _, x := range nums {
		mp[x] = true
	}
	ans := 0
	for _, x := range nums {
		if mp[x+diff] && mp[x+2*diff] {
			ans++
		}
	}
	return ans
}

File: LeetCode/Simulation/JS/Leetcode2367-number-of-arithmetic-triplets.js
var arithmeticTriplets = function (nums, diff) {
  let st = new Set();
  for (const x of nums) {
    st.add(x);
  }
  let ans = 0;
  for (const x of nums) {
    if (st.has(x + diff) && st.has(x + 2 * diff)) {
      ans++;
    }
  }
  return ans;
};